crates/matrix-sdk-ui/src/room_list_service/room.rs

```diff
@@ -137,13 +137,27 @@ impl Room {
             .clone()
     }
 
-    /// Get the latest event in the timeline, if we already have it cached. Does
-    /// not fetch any events or calculate anything - if it's not already
-    /// available, we return None.
+    /// Get the latest event in the timeline.
+    ///
+    /// The latest event comes first from the `Timeline` if it exists.
+    /// Otherwise, it comes from the cache. This method does not fetch any
+    /// events or calculate anything — if it's not already available, we
+    /// return `None`.
```

Comment:
I don't think this is right, since we decided that (at least eventually) the latest event could also be something that's not a timeline item – notably a reaction to the user's message if it's otherwise the latest. So I think we should only use the `latest_event` from the timeline if that's a local echo. Or go back to the design stage and decide how local echoes are really supposed to interact with everything else, because I don't think it has been specified very clearly.

Reply:
So it's just a matter of removing the last commit of this patch, https://github.com/matrix-org/matrix-rust-sdk/pull/2400/commits/2290bae9425a9b3c3d67b9b560e6389f410da515. I was anticipating this comment.

Reply:
Yes :D